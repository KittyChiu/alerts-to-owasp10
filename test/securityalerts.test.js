// securityalerts.test.js
const getSecurityAlerts = require('../src/securityalerts')
const fs = require('fs').promises // Import the promises version of fs
require('dotenv').config()

describe('securityalerts', () => {
  it('should extract alerts and match snapshot', async () => {
    const result = await getSecurityAlerts(
      process.env.ORGANISATION,
      process.env.GITHUB_TOKEN
    )

    // Write the result to a file
    await fs.writeFile('./alerts.json', JSON.stringify(result, null, 2))

    expect(result).toMatchSnapshot()
  })
})
