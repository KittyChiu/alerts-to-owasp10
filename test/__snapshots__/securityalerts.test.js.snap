// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`securityalerts should extract alerts and match snapshot 1`] = `
{
  "codeql-nodegoat-template": {
    "1": [
      "cwe-1333",
      "cwe-400",
      "cwe-730",
    ],
    "10": [
      "cwe-079",
      "cwe-116",
    ],
    "11": [
      "cwe-079",
      "cwe-116",
    ],
    "12": [
      "cwe-079",
      "cwe-116",
    ],
    "13": [
      "cwe-079",
      "cwe-116",
    ],
    "14": [
      "cwe-079",
      "cwe-116",
    ],
    "15": [
      "cwe-079",
      "cwe-116",
    ],
    "16": [
      "cwe-079",
      "cwe-116",
    ],
    "17": [
      "cwe-079",
      "cwe-116",
    ],
    "18": [
      "cwe-079",
      "cwe-116",
    ],
    "19": [
      "cwe-079",
      "cwe-116",
    ],
    "2": [
      "cwe-1333",
      "cwe-400",
      "cwe-730",
    ],
    "20": [
      "cwe-079",
      "cwe-116",
    ],
    "21": [
      "cwe-079",
      "cwe-094",
      "cwe-095",
      "cwe-116",
    ],
    "22": [
      "cwe-079",
      "cwe-094",
      "cwe-095",
      "cwe-116",
    ],
    "23": [
      "cwe-079",
      "cwe-094",
      "cwe-095",
      "cwe-116",
    ],
    "24": [
      "cwe-079",
      "cwe-094",
      "cwe-095",
      "cwe-116",
    ],
    "25": [
      "cwe-601",
    ],
    "26": [
      "cwe-311",
      "cwe-312",
      "cwe-319",
      "cwe-614",
    ],
    "3": [
      "cwe-1333",
      "cwe-400",
      "cwe-730",
    ],
    "4": [
      "cwe-307",
      "cwe-400",
      "cwe-770",
    ],
    "5": [
      "cwe-089",
      "cwe-090",
      "cwe-943",
    ],
    "6": [
      "cwe-089",
      "cwe-090",
      "cwe-943",
    ],
    "7": [
      "cwe-352",
    ],
    "8": [
      "cwe-918",
    ],
    "9": [
      "cwe-079",
      "cwe-116",
    ],
  },
  "webgoat-demo-2": {
    "1": [
      "cwe-079",
      "cwe-116",
    ],
    "10": [
      "cwe-020",
      "cwe-080",
      "cwe-116",
    ],
    "11": [
      "cwe-020",
      "cwe-080",
      "cwe-116",
    ],
    "12": [
      "cwe-020",
    ],
    "2": [
      "cwe-079",
      "cwe-116",
    ],
    "3": [
      "cwe-079",
      "cwe-116",
    ],
    "4": [
      "cwe-079",
      "cwe-116",
    ],
    "5": [
      "cwe-079",
      "cwe-116",
    ],
    "6": [
      "cwe-079",
      "cwe-116",
    ],
    "7": [
      "cwe-079",
      "cwe-116",
    ],
    "8": [
      "cwe-079",
      "cwe-116",
    ],
    "9": [
      "cwe-079",
      "cwe-116",
    ],
  },
  "webgoat-demo-mb": {
    "1": [
      "cwe-079",
      "cwe-116",
    ],
    "10": [
      "cwe-020",
      "cwe-080",
      "cwe-116",
    ],
    "11": [
      "cwe-020",
      "cwe-080",
      "cwe-116",
    ],
    "2": [
      "cwe-079",
      "cwe-116",
    ],
    "3": [
      "cwe-079",
      "cwe-116",
    ],
    "4": [
      "cwe-079",
      "cwe-116",
    ],
    "5": [
      "cwe-079",
      "cwe-116",
    ],
    "6": [
      "cwe-079",
      "cwe-116",
    ],
    "7": [
      "cwe-079",
      "cwe-116",
    ],
    "8": [
      "cwe-079",
      "cwe-116",
    ],
    "9": [
      "cwe-079",
      "cwe-116",
    ],
  },
}
`;
